From 7e9980940718642509dfea63be4cd14cf309602d Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Ball=C3=B3=20Gy=C3=B6rgy?= <ballogyor@gmail.com>
Date: Wed, 24 Jan 2024 19:33:29 +0100
Subject: [PATCH] autostart: add launcher for Geoclue Demo agent

Geoclue client applications need a Geoclue agent to work, which is autostarted with the session. Gnome-shell contains an internal Geoclue agent, while the Geoclue package provides a demo agent with an autostart file with NotShowIn=GNOME. This causes that the agent is available in all desktop environments except in GNOME Flashback.

This commit adds an autostart file for the demo agend just for GNOME Flashback.

Alternative solution would be to implement a Geoclue agent within GNOME Flashback.
---
 data/autostart/Makefile.am                            |  4 ++++
 .../gnome-flashback-geoclue-demo-agent.desktop.in.in  | 11 +++++++++++
 po/POTFILES.in                                        |  1 +
 3 files changed, 16 insertions(+)
 create mode 100644 data/autostart/gnome-flashback-geoclue-demo-agent.desktop.in.in

diff --git a/data/autostart/Makefile.am b/data/autostart/Makefile.am
index d31d1c2..c0cbdee 100644
--- a/data/autostart/Makefile.am
+++ b/data/autostart/Makefile.am
@@ -3,6 +3,7 @@ NULL =
 autostartdir = $(sysconfdir)/xdg/autostart
 autostart_DATA = \
 	gnome-flashback-clipboard.desktop \
+	gnome-flashback-geoclue-demo-agent.desktop \
 	gnome-flashback-idle-monitor.desktop \
 	gnome-flashback-media-keys.desktop \
 	gnome-flashback-nm-applet.desktop \
@@ -18,6 +19,7 @@ autostart_DATA = \
 
 EXTRA_DIST = \
 	gnome-flashback-clipboard.desktop.in.in \
+	gnome-flashback-geoclue-demo-agent.desktop.in.in \
 	gnome-flashback-idle-monitor.desktop.in.in \
 	gnome-flashback-media-keys.desktop.in.in \
 	gnome-flashback-nm-applet.desktop.in \
@@ -27,6 +29,8 @@ EXTRA_DIST = \
 CLEANFILES = \
 	gnome-flashback-clipboard.desktop.in \
 	gnome-flashback-clipboard.desktop \
+	gnome-flashback-geoclue-demo-agent.desktop.in \
+	gnome-flashback-geoclue-demo-agent.desktop \
 	gnome-flashback-idle-monitor.desktop.in \
 	gnome-flashback-idle-monitor.desktop \
 	gnome-flashback-media-keys.desktop.in \
diff --git a/data/autostart/gnome-flashback-geoclue-demo-agent.desktop.in.in b/data/autostart/gnome-flashback-geoclue-demo-agent.desktop.in.in
new file mode 100644
index 0000000..52cd976
--- /dev/null
+++ b/data/autostart/gnome-flashback-geoclue-demo-agent.desktop.in.in
@@ -0,0 +1,11 @@
+[Desktop Entry]
+Name=Geoclue Demo agent (GNOME Flashback)
+GenericName=Demo geoclue agent
+Keywords=geolocation;
+Exec=@libexecdir@/geoclue-2.0/demos/agent
+TryExec=@libexecdir@/geoclue-2.0/demos/agent
+Icon=mark-location-symbolic
+OnlyShowIn=GNOME-Flashback;
+NoDisplay=true
+Terminal=false
+Type=Application
diff --git a/po/POTFILES.in b/po/POTFILES.in
index cf61d75..4db18df 100644
--- a/po/POTFILES.in
+++ b/po/POTFILES.in
@@ -9,6 +9,7 @@ daemons/polkit/flashback-polkit-dialog.c
 daemons/polkit/flashback-polkit-dialog.ui
 data/applications/gnome-flashback.desktop.in.in
 data/autostart/gnome-flashback-clipboard.desktop.in.in
+data/autostart/gnome-flashback-geoclue-demo-agent.desktop.in.in
 data/autostart/gnome-flashback-idle-monitor.desktop.in.in
 data/autostart/gnome-flashback-media-keys.desktop.in.in
 data/autostart/gnome-flashback-nm-applet.desktop.in
-- 
2.43.0

